junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/descriptor/LifecycleMethodUtilsTest.java

```diff
@@ -0,0 +1,223 @@
+		JUnitException exception = assertThrows(JUnitException.class,
+			() -> findAfterEachMethods(TestCaseWithNonVoidLifecyleMethods.class));
+		assertEquals(
+			"@AfterEach method 'int org.junit.jupiter.engine.descriptor.TestCaseWithNonVoidLifecyleMethods.bb()' must not return a value.",
+			exception.getMessage());
+	}
+
+	@Test
+	void findBeforeEachMethodsWithStandardLifecycle() {
+		List<Method> methods = findBeforeEachMethods(TestCaseWithStandardLifecycle.class);
+
+		assertEquals(2, methods.size());
+		assertTrue(contains(methods, "nine"));
+		assertTrue(contains(methods, "ten"));
+	}
+
+	@Test
+	void findAfterEachMethodsWithStandardLifecycle() {
+		List<Method> methods = findAfterEachMethods(TestCaseWithStandardLifecycle.class);
+
+		assertEquals(2, methods.size());
+		assertTrue(contains(methods, "eleven"));
+		assertTrue(contains(methods, "twelve"));
+	}
+
+	@Test
+	void findBeforeAllMethodsWithStandardLifecycleAndWithoutRequiringStatic() {
+		List<Method> methods = findBeforeAllMethods(TestCaseWithStandardLifecycle.class, false);
+
+		assertEquals(1, methods.size());
+		assertTrue(contains(methods, "one"));
+	}
+
+	@Test
+	void findBeforeAllMethodsWithStandardLifecycleAndRequiringStatic() {
+		JUnitException exception = assertThrows(JUnitException.class,
+			() -> findBeforeAllMethods(TestCaseWithStandardLifecycle.class, true));
+		assertEquals(
+			"@BeforeAll method 'void org.junit.jupiter.engine.descriptor.TestCaseWithStandardLifecycle.one()' must be static unless the test class is annotated with @TestInstance(Lifecycle.PER_CLASS).",
+			exception.getMessage());
+	}
+
+	@Test
+	void findBeforeAllMethodsWithLifeCyclePerClassAndRequiringStatic() {
+		List<Method> methods = findBeforeAllMethods(TestCaseWithLifecyclePerClass.class, false);
+
+		assertEquals(2, methods.size());
+		assertTrue(contains(methods, "three"));
+		assertTrue(contains(methods, "four"));
+	}
+
+	@Test
+	void findAfterAllMethodsWithStandardLifecycleAndWithoutRequiringStatic() {
+		List<Method> methods = findAfterAllMethods(TestCaseWithStandardLifecycle.class, false);
+
+		assertEquals(2, methods.size());
+		assertTrue(contains(methods, "five"));
+		assertTrue(contains(methods, "six"));
+	}
+
+	@Test
+	void findAfterAllMethodsWithStandardLifecycleAndRequiringStatic() {
+		assertThrows(JUnitException.class, () -> findAfterAllMethods(TestCaseWithStandardLifecycle.class, true));
+	}
+
+	@Test
+	void findAfterAllMethodsWithLifeCyclePerClassAndRequiringStatic() {
+		List<Method> methods = findAfterAllMethods(TestCaseWithLifecyclePerClass.class, false);
+
+		assertEquals(2, methods.size());
+		assertTrue(contains(methods, "seven"));
+		assertTrue(contains(methods, "eight"));
+	}
+
+	private boolean contains(List<Method> methods, String methodName) {
+		// @formatter:off
+        return methods.stream()
+                .map(Method::getName)
+                .anyMatch(name -> name.equals(methodName));
+        // @formatter:on
+	}
```

Comment:
@jbduncan Good call! It was a tabs vs spaces issue which got obscured by turning off the formatter...